go.mod

```diff
@@ -3,6 +3,8 @@ module github.com/celestiaorg/go-cnc
 go 1.19
 
 require (
+	cosmossdk.io/errors v1.0.0-beta.7
+	github.com/cosmos/cosmos-sdk v0.47.2
```

Comment:
We need to confirm with @tzdybal whether we can import cosmos or not, because of the "dependency hell"

Reply:
I've looked into option to extract required code from cosmos to remove new dependancies, but it seems to be very complicated.

Reply:
Ah, also this repo still not on go1.20 and don't have access to nice `errors.Join`. @tzdybal  would you mind if I upgrade to golang versio?

Reply:
We cannot import cosmos sdk i think that's why go-cnc exists. We also are not running on 47.

Reply:
The alternative option would be to print an error on node side and transport it as json field over the http.

Reply:
The replace directive in go.mod will use the specific fork of cosmossdk Rollkit uses, so technically it should be able to import, unless there is some other subtle detail we miss.

Reply:
We need to support multiple cosmos sdk forks so this will break.

Reply:
You can't have multiple cosmos-sdk forks/replaces within a single binary.

Reply:
Since we are stuck here, I propose to just remove cosmos-sdk and resolve this

Reply:
I've updated PR to use ad-hoc encoding instead of original cosmos-sdk one. It allows go-cnc to not rely on cosmos sdk dependancy at all. 
 Keep in mind, the previous API was not compatible with cosmos sdk `error.Is`  support, so nothing really changes on error handling level, except the error now will have a bit different templating.